carddav/server.go

```diff
@@ -614,43 +615,160 @@ func (b *backend) propFindAllAddressObjects(ctx context.Context, propfind *inter
 }
 
 func (b *backend) PropPatch(r *http.Request, update *internal.PropertyUpdate) (*internal.Response, error) {
-	homeSetPath, err := b.Backend.AddressBookHomeSetPath(r.Context())
-	if err != nil {
-		return nil, err
-	}
-
+	resType := b.resourceTypeAtPath(r.URL.Path)
 	resp := internal.NewOKResponse(r.URL.Path)
 
-	if r.URL.Path == homeSetPath {
-		// TODO: support PROPPATCH for address books
+	switch resType {
+	case resourceTypeAddressBook:
+		ab, err := b.Backend.GetAddressBook(r.Context(), r.URL.Path)
```

Comment:
Hm, this is not atomic… If the address book is updated by another client after `GetAddressBook`, but before `UpdateAddressBook`, then we'll be passing stale data.

Can we instead indicate somehow which fields are being updated? This could be done by passing a different struct with pointers (nil would mean "leave as is"), or by passing a bitmask of updated fields, for instance.